projects/ngx-datatable/src/lib/components/datatable.component.ts

```diff
@@ -846,8 +854,22 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit, After
     if (!columns) {return undefined;}
 
     let width = this._innerWidth;
-    if (this.scrollbarV) {
+    if (this.scrollbarV && !this.scrollbarVDynamic) {
       width = width - this.scrollbarHelper.width;
+
+    } else if (this.scrollbarVDynamic){
+      const scrollerHeight = this.bodyComponent?.scroller?.element.offsetHeight;
+      if (scrollerHeight && this.bodyHeight < scrollerHeight) {
+        width = width - this.scrollbarHelper.width;
+      }
+
+      if (this.headerComponent && this.headerComponent.innerWidth !== width){
```

Comment:
You've already used *optional chaining* above, so let's also do this here:

```suggestion
      if (this.headerComponent?.innerWidth !== width){
```